src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -91,73 +90,116 @@ public void Dictionary_Generic_RemoveKey_DefaultKeyContainedInDictionary(int cou
         public void EnsureCapacity_NegativeCapacityRequested_Throws()
         {
             var dictionary = new Dictionary<string, string>();
-            Assert.Throws<ArgumentOutOfRangeException>(() => dictionary.EnsureCapacity(-1));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => dictionary.EnsureCapacity(-1));
+        }
+
+        [Fact]
+        [ActiveIssue("dotnet/corefx #24445")] 
+        public void EnsureCapacity_DictionaryNotInitialized_RequestedZero_ReturnsZero()
+        {
+            var dictionary = new Dictionary<string, string>();
+            Assert.Equal(0, dictionary.EnsureCapacity(0));
         }
 
         [Theory]
-        [InlineData(0)]
         [InlineData(1)]
+        [InlineData(2)]
         [InlineData(3)]
-        public void EnsureCapacity_DefaultCapacityOnEmptyDictionary_ReturnsCapacityRequestedLargerOrEqual(int requestedCapacity)
+        [InlineData(4)]
+        public void EnsureCapacity_DictionaryNotInitialized_RequestedNonZero_CapacityIsSetToAtLeastTheRequested(int requestedCapacity)
         {
             var dictionary = new Dictionary<string, string>();
             Assert.InRange(dictionary.EnsureCapacity(requestedCapacity), requestedCapacity, int.MaxValue);
         }
 
         [Theory]
-        [InlineData(1)]
         [InlineData(3)]
-        public void EnsureCapacity_CapacityRequestedSmallerThanCurrentCapacity_CapacityUnchanged(int requestedCapacity)
+        [InlineData(7)]
+        public void EnsureCapacity_RequestedCapacitySmallerThanCurrent_CapacityUnchanged(int currentCapacity)
         {
-            var dictionary = new Dictionary<string, string>();
-            var capacity = dictionary.EnsureCapacity(requestedCapacity);
-            Assert.Equal(capacity, dictionary.EnsureCapacity(requestedCapacity - 1));
+            Dictionary<string, string> dictionary;
+
+            // assert capacity remains the same when ensuring a capacity smaller or equal to existing
+            for (int i = 0; i <= currentCapacity; i++)
+            {
+                dictionary = new Dictionary<string, string>(currentCapacity);
+                Assert.Equal(currentCapacity, dictionary.EnsureCapacity(i));
+            }
+        }
+
+        [Fact]
+        public void EnsureCapacity_ExistingCapacityRequested_SameValueReturned()
+        {
+            int capacity = 7;
```

Comment:
nit: `const int Capacity = 7;`

Reply:
I passed 7 down to the test as InlineData instead.